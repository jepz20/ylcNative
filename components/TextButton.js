// @flow

import * as React from 'react'
import { Text, StyleSheet } from 'react-native'
import { Button } from './Button'

type Props = {
  size?: 'large' | 'small' | 'medium',
  value: string,
  onPress: any
}
const TextButton: React.StatelessFunctionalComponent<Props> = ({
  size = 'medium',
  value,
  ...props
}) => {
  return (
    <Button {...props}>
      <Text style={[styles.text, styles[size]]}>{value}</Text>
    </Button>
  )
}

const styles = StyleSheet.create({
  text: {
    color: '#fff',
    fontSize: 40
  },
  small: {
    fontSize: 15
  },
  large: {
    fontSize: 100
  },
  medium: {
    fontSize: 40
  }
})

export { TextButton }
