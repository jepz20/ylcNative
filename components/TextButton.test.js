import 'react-native'
import React from 'react'
import { TextButton } from './TextButton'

import renderer from 'react-test-renderer'

describe('TextButton', () => {
  test('renders with no props', () => {
    const textButton = renderer.create(<TextButton />).toJSON()
    expect(textButton).toMatchSnapshot()
  })

  test('renders with a text', () => {
    const textButton = renderer.create(<TextButton value={'test'} />).toJSON()
    expect(textButton).toMatchSnapshot()
  })

  test('renders a small button with a text', () => {
    const textButton = renderer
      .create(<TextButton size='small' value={'test'} />)
      .toJSON()
    expect(textButton).toMatchSnapshot()
  })

  test('renders a medium button with a text', () => {
    const textButton = renderer
      .create(<TextButton size='medium' value={'test'} />)
      .toJSON()
    expect(textButton).toMatchSnapshot()
  })

  test('renders a large button with a text', () => {
    const textButton = renderer
      .create(<TextButton size='large' value={'test'} />)
      .toJSON()
    expect(textButton).toMatchSnapshot()
  })
})
