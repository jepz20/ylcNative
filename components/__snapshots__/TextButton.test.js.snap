// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextButton renders a large button with a text 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
      ],
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#fff",
          "fontSize": 40,
        },
        Object {
          "fontSize": 100,
        },
      ]
    }
  >
    test
  </Text>
</View>
`;

exports[`TextButton renders a medium button with a text 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
      ],
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#fff",
          "fontSize": 40,
        },
        Object {
          "fontSize": 40,
        },
      ]
    }
  >
    test
  </Text>
</View>
`;

exports[`TextButton renders a small button with a text 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
      ],
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#fff",
          "fontSize": 40,
        },
        Object {
          "fontSize": 15,
        },
      ]
    }
  >
    test
  </Text>
</View>
`;

exports[`TextButton renders with a text 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
      ],
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#fff",
          "fontSize": 40,
        },
        Object {
          "fontSize": 40,
        },
      ]
    }
  >
    test
  </Text>
</View>
`;

exports[`TextButton renders with no props 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
      ],
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#fff",
          "fontSize": 40,
        },
        Object {
          "fontSize": 40,
        },
      ]
    }
  />
</View>
`;
